### PR TITLE
prl-tools: 17.1.4-51567 -> 18.0.0-53049

### PR DESCRIPTION
**Pull request by catap**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by wegank — CHANGES_REQUESTED** [review]
The patches are no longer required on aarch64. Could you please remove them?

P.-S. Is this the same case on x86_64?

**Comment by catap** [comment]
@wegank I'm testing it right now on x86_64; will be back as soon as it finish rebuilding everything.

**Review by wegank — APPROVED** [review]

File: pkgs/os-specific/linux/prl-tools/default.nix [hunk 1]
```diff
@@ -56,8 +56,8 @@ stdenv.mkDerivation rec {
     fi
   '';
 
-  patches = lib.optional (lib.versionAtLeast kernel.version "5.18") ./prl-tools-5.18.patch
-    ++ lib.optional (lib.versionAtLeast kernel.version "5.19") ./prl-tools-5.19.patch;
+  # patches = lib.optional (lib.versionAtLeast kernel.version "5.18") ./prl-tools-5.18.patch
+  #   ++ lib.optional (lib.versionAtLeast kernel.version "5.19") ./prl-tools-5.19.patch;
```
Comment (mweinelt) [inline pkgs/os-specific/linux/prl-tools/default.nix:60]:
```suggestion
```

Remove these lines if they don't apply anymore.

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Review by wegank — APPROVED** [review]

**Comment by sandangel** [comment]
nice, I was looking for this. thanks for making the PR.

File: pkgs/os-specific/linux/prl-tools/default.nix [hunk 2]
```diff
@@ -25,14 +25,14 @@ assert (!libsOnly) -> kernel != null;
 assert lib.elem stdenv.hostPlatform.system [ "x86_64-linux" "i686-linux" "aarch64-linux" ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/prl-tools/default.nix:25]:
```suggestion
```
This should bet set in meta.platforms only and not prevent evaluation on other platforms.

**Comment by catap** [comment]
> nice, I was looking for this. thanks for making the PR. Does anyone experience weird issue on keyboard? It seems more sticky than version 17. E.g. Many characters was entered when I hit only 1 char. I was using version 18 with the prl-tools from ver 17. maybe that is why. Hope this PR would also fix that.

I may confirm that I have the same wired input into macOS 10.5 without any Parallels Tools :( Seems it is a bug.

**Comment by catap** [comment]
@SuperSandro2000 it's done

**Comment by sandangel** [comment]
can we delay this PR until we can see no more issue? Probably after another kernel/paralell update

**Comment by wegank** [comment]
@sandangel Does the issue persist when Parallels Tools are not installed, as in the case of @catap?

Personally, I'm not experiencing those issues on my laptop, and I would like to see a minimal configuration that reproduces the issue. I guess it could be related to locales, i.e. using `en_US` instead of `en_US.UTF-8`.

**Comment by sandangel** [comment]
let me investigate on my side a bit

**Comment by catap** [comment]
> can we delay this PR until we can see no more issue? Probably after another kernel/paralell update

@sandangel I suggest to merge ASAP and address an issue when we have it ;)

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by sandangel** [comment]
sound good to me too.

**Review by sandangel — APPROVED** [review]

**Pull request merged**